systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectST.java

```diff
@@ -1238,6 +1239,79 @@ void testConfigureDeploymentStrategy(ExtensionContext extensionContext) {
+                .editSpec()
+                .editKafka()
+                .withListeners(new GenericKafkaListenerBuilder()
+                        .withName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+                        .withPort(9092)
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withTls(false)
+                        .withAuth(new KafkaListenerAuthenticationScramSha512())
+                        .build())
+                .endKafka()
+                .endSpec()
+                .build());
+
+        KafkaUser kafkaUser =  KafkaUserTemplates.scramShaUser(clusterName, userName).build();
+
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(false, kafkaClientsName).build());
+        resourceManager.createResource(extensionContext, KafkaUserTemplates.scramShaUser(clusterName, userName).build());
+        resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(clusterName, topicName).build());
+        resourceManager.createResource(extensionContext, KafkaConnectTemplates.kafkaConnect(extensionContext, clusterName, 1)
+                .withNewSpec()
+                    .withBootstrapServers(KafkaResources.plainBootstrapAddress(clusterName))
+                    .withNewKafkaClientAuthenticationScramSha512()
+                        .withUsername(userName)
+                        .withPasswordSecret(new PasswordSecretSourceBuilder()
+                            .withSecretName(userName)
+                            .withPassword("password")
+                            .build())
+                    .endKafkaClientAuthenticationScramSha512()
+                    .addToConfig("key.converter.schemas.enable", false)
+                    .addToConfig("value.converter.schemas.enable", false)
+                    .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                    .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+                    .withVersion(Environment.ST_KAFKA_VERSION)
+                    .withReplicas(1)
+                .endSpec()
+                .build());
+
+        final String kafkaConnectPodName = kubeClient(namespaceName).listPodsByPrefixInName(KafkaConnectResources.deploymentName(clusterName)).get(0).getMetadata().getName();
+
+        KafkaConnectUtils.waitUntilKafkaConnectRestApiIsAvailable(namespaceName, kafkaConnectPodName);
+
+        Map<String, String> connectSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaConnectResources.deploymentName(clusterName));
+        String newPassword = "bmVjb0ppbmVob05lelNwcmF2bnlQYXNzd29yZA==";
+        Secret passwordSecret = new SecretBuilder()
+                .withNewMetadata()
+                    .withName(userName)
+                .endMetadata()
+                .addToData("password", newPassword)
+                .build();
```

Comment:
I think it would be cleaner to use a custom password and provide it to the `KAfkaUSer` using the (new) feature for it and not just rewrite the generated user secret.